R/derive_joined.R

```diff
@@ -241,24 +270,54 @@
 #'   dataset_add = windows,
 #'   join_type = "all",
 #'   filter_join = AWLO <= ADY & ADY <= AWHI
-#' )
+#' ) %>%
+#'   select(USUBJID, ADY, AWLO, AWHI, AVISIT)
+#'
+#' @caption Join only the lowest/highest value occurring within a condition (`filter_join`,
+#'   `order` and `mode`)
+#' @info Derive the nadir value for each observation (i.e. the lowest value
+#'   occurring before) by subject.
+#'
+#' - Note how `dataset` and `dataset_add` are the same here, so we are joining
+#'   a dataset with itself. This enables us to compare records within the dataset
+#'   to each other.
+#' - Now we use `by_vars` as we only want to perform the join by subject.
+#' - To find the lowest value we use the `order` and `mode` arguments.
+#' - We subsequently need to check `ADY` to only check assessments occurring
+#'   before. As this is not included in `by_vars` or `order`, we have to ensure
+#'   it also gets joined by adding to `join_vars`. Then in `filter_join` note
+#'   how `ADY.join < ADY` is used as the same variable exists in both datasets,
+#'   so the version from `dataset_add` has `.join` added.
+#' - Finally, we use `check_type = "none"` because this is a case where there
+#'   could be duplicates according to the sort order used (e.g. see subject `"1"`
+#'   records at day 1 and 8). Given the values are the same it doesn't matter
+#'   to us which exact one is taken, so we silence the default warning here.
+#' @code
+#' derive_vars_joined(
+#'   adbds,
+#'   dataset_add = adbds,
+#'   by_vars = exprs(STUDYID, USUBJID),
+#'   order = exprs(AVAL),
+#'   new_vars = exprs(NADIR = AVAL),
+#'   join_vars = exprs(ADY),
+#'   join_type = "all",
+#'   filter_join = ADY.join < ADY,
+#'   mode = "first",
+#'   check_type = "none"
+#' ) %>%
+#'   select(USUBJID, ADY, AVAL, NADIR)
 #'
-#' # derive the nadir after baseline and before the current observation
-#' adbds <- tribble(
-#'   ~USUBJID, ~ADY, ~AVAL,
-#'   "1",        -7,    10,
-#'   "1",         1,    12,
-#'   "1",         8,    11,
-#'   "1",        15,     9,
-#'   "1",        20,    14,
-#'   "1",        24,    12,
-#'   "2",        13,     8
-#' )
+#' @caption Filtering which records are joined from the additional dataset (`filter_add`)
+#' @info Imagine we wanted to achieve the same as above, but we now want to derive
+#'   this allowing only post-baseline values to be possible for the nadir.
```

Comment:
```suggestion
#'   this allowing only post-baseline values to be possible for the lowest value.
```

dang who went to the fancy college?

Reply:
see comment above